alacritty_terminal/src/display.rs

```diff
@@ -521,14 +521,16 @@ impl Display {
                     // Iterate over all non-empty cells in the grid
                     for cell in grid_cells {
                         // Update underline/strikeout
-                        rects.update_lines(&cell, &size_info, &metrics);
+                        lines.update_lines(&cell);
 
                         // Draw the cell
                         api.render_cell(cell, glyph_cache);
                     }
                 });
             }
 
+            let mut rects = Rects::from(lines.into_rects(&metrics, &size_info));
```

Comment:
I think `rects` should just be `Vec<Rect>` and `Line::into_rects` should return `Vec<Rect>`. I think that should work.

I just saw that `Rects` obviously has colors too, but I don't think we use `Rect` anywhere else, right? So we could probably just add the color to `Rect` instead?

Reply:
So, we're changing renderer? I'm fine with it though.

Reply:
Oh, and in this case we don't actually need `Rects` any more.

Reply:
I've checked the use of `Rect` struct and we can't add color to it, without reworking all the stuff.

Reply:
> I've checked the use of Rect struct and we can't add color to it, without reworking all the stuff.

What do you mean by `all the stuff`, that sounds quite dramatic. As far as I can tell, wed just have to merge the function parameters in the renderer everywhere where it takes `rect` and `color` and have it both in one, right?

I don't think there's any instance of `Rect` without `color`, is there? So that kinda sounds like it would almost be an improvement anyways.

My main concern is that `Rect` sounds like a very generic name and we couldn't use it anymore if we ever wanted to have just a normal rectangle without any colors not used in the renderer, but I don't think that's currently the case anywhere?

Reply:
> I don't think there's any instance of Rect without color, is there?

I've accidentally found another `Rect` struct, which isn't related to `Rect` in renderer. So, I'll change stuff in a moment.

Reply:
> I've accidentally found another Rect struct, which isn't related to Rect in renderer. So, I'll change stuff in a moment.

There is? I don't see any other `Rect` struct in the latest master.

Reply:
I was grepping entire workspace, so there is one in `font/src/darwin/mod.rs`, which is unrelated here.